app/controllers/assessment_questions_controller.rb

```diff
@@ -11,9 +11,39 @@ def your_risk_post
 
     end
 
-    (session['current_assessment'] ||= Hash.new)[:confidence_level_required] = params[:confidence_level_required]
+    get_assessment()['confidence_level_required'] = params[:confidence_level_required]
 
     redirect_to controller: 'assessments', action: 'overview'
   end
 
+  def choose_evidence_get
+    render "assessments/choose-evidence"
+  end
+
+  def choose_evidence_post
+    if (not params[:evidence_type]) && (params[:evidence_type_other].blank?)
+      @errors[:evidence_type] = 'You must choose a piece of evidence'
+      render "assessments/choose-evidence" and return
+    end
+    if params[:id] == "new"
+      id = SecureRandom.uuid
+    else
+      id = params[:id]
+    end
```

Comment:
What does this `id` parameter get used for? I can't see anything that reads it?

Reply:
We're storing evidence with the ID as a key so that (in a future story) we can delete the evidence again, and know which evidence the user wants to delete. (And potentially in future edit it, maybe.) So the the id is the key in `get_evidence`, below, and it gets stored as part of the assessment.

Reply:
👍